pkg/controllers/cidrs_controller.go

```diff
@@ -103,25 +106,130 @@ func (r *CIDRReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 	return ctrl.Result{}, nil
 }
 
+// trimSpaces removes leading and trailing spaces from each string in the slice
+func trimSpaces(values []string) []string {
+	result := make([]string, 0, len(values))
+	for _, v := range values {
+		trimmed := strings.TrimSpace(v)
+		if trimmed != "" {
+			result = append(result, trimmed)
+		}
+	}
+	return result
+}
+
 func applyProcessor(reader io.Reader, processing ipamv1alpha1.Processing) ([]string, error) {
-	if processing.JSONPath != "" {
+	// Default to YAML format if not specified (backward compatibility)
+	format := processing.Format
+	if format == "" {
+		format = ipamv1alpha1.YAML
+	}
+
+	switch format {
+	// Handle simple text formats CSV
+	case ipamv1alpha1.CommaSeparatedValues:
+		data := csv.NewReader(reader)
+		records, err := data.Read()
+		if err != nil {
+			return nil, fmt.Errorf("failed to read comma-separated values: %w", err)
+		}
+		return trimSpaces(records), nil
+
+	// Handle simple text formats LSV
+	case ipamv1alpha1.LineSeparatedValues:
+		data, err := io.ReadAll(reader)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read line-separated values: %w", err)
+		}
+
+		lines := trimSpaces(strings.Split(string(data), "\n"))
+
+		var values []string
+		for _, line := range lines {
+			if line != "" && !strings.HasPrefix(line, "#") { // Skip empty lines and comments
+				values = append(values, line)
+			}
+		}
+
+		return values, nil
+
+	// Handle complex text formats
+	// * combined - when CVS contains coma and lines and descriptions
+	// * simple text between html/xml code
+	// * buffered so slower - not recommended, only edge cases
+	case ipamv1alpha1.CombinedSeparatedValues:
```

Comment:
Do we have any example of those? (i.e. providers exposing their IPs exclusively through this?

Reply:
[That's](https://techdocs.akamai.com/property-manager/pdfs/akamai_ipv4_CIDRs.csv) a pure CSV (well LSV), right?

It seems like solving a problem we don't have.
I would not have it and wait to scope the problem to solvebefore implementation

Reply:
correct, but any other sophisticated (description)  CSV would not work :/